backend/src/main/java/com/bakdata/conquery/models/preproc/Preprocessor.java

```diff
@@ -93,9 +89,10 @@ public static File getTaggedVersion(File file, String tag, String extension) {
 	 *
 	 * Reads CSV file, per row extracts the primary key, then applies other transformations on each row, then compresses the data with {@link ColumnStore}.
 	 */
-	public static void preprocess(TableImportDescriptor descriptor, ProgressBar totalProgress, ConqueryConfig config) throws IOException {
+	public static void preprocess(PreprocessingJob preprocessingJob, ProgressBar totalProgress, ConqueryConfig config) throws IOException {
 
-		final File preprocessedFile = descriptor.getInputFile().getPreprocessedFile();
+		final File preprocessedFile = preprocessingJob.getPreprocessedFile();
+		TableImportDescriptor descriptor = preprocessingJob.getDescriptor();
 
 		// Create temp file that will be moved when finished (we ensure the same file system, to avoid unnecessary copying)
```

Comment:
Ich finde die Abgrenzung zwischen Preprocess und PreprocessJob ziemlich blurry, weil der Job hier ausgepackt wird und statt wie die ConqueryJobs selber die Arbeit verrichten oder deligieren.

Reply:
Ich wollte eigentlich PreprocessingJob aus der Funktion ganz holen, aber weil man die echten Dateinamen erst resolven muss, wird das noch benötigt. Davor haben wir in die Descriptions geschrieben, was ich nicht gut finde.